waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -56,54 +63,73 @@ finalizeMigration genProjectRootDirAbs dbMigrationsDirInWaspProjectDirAbs onLast
-  do
-    copyDirectoryRecursive genProjectMigrationsDir waspMigrationsDir >> return (Right ())
-      `catch` (\e -> return $ Left $ show (e :: P.PathException))
-      `catch` (\e -> return $ Left $ show (e :: IOError))
+  copyDirectoryRecursive genProjectMigrationsDir waspMigrationsDir >> return (Right ())
+    `catch` (\e -> return $ Left $ show (e :: P.PathException))
+    `catch` (\e -> return $ Left $ show (e :: IOError))
   where
-    waspMigrationsDir = SP.castDir dbMigrationsDirInWaspProjectDirAbs
-    genProjectMigrationsDir = SP.castDir $ genProjectRootDirAbs SP.</> dbRootDirInProjectRootDir SP.</> dbMigrationsDirInDbRootDir
+    waspMigrationsDir = dbMigrationsDirInWaspProjectDirAbs
+    genProjectMigrationsDir = genProjectRootDirAbs </> dbRootDirInProjectRootDir </> dbMigrationsDirInDbRootDir
 
 -- | This function assumes the DB schema has been generated, as it will attempt to read it from the generated code.
-writeDbSchemaChecksumToFile :: Path' Abs (Dir ProjectRootDir) -> Path' (Rel ProjectRootDir) File' -> IO ()
+writeDbSchemaChecksumToFile ::
+  DbSchemaChecksumFile f =>
+  Path' Abs (Dir ProjectRootDir) ->
+  Path' (Rel ProjectRootDir) (File f) ->
+  IO ()
 writeDbSchemaChecksumToFile genProjectRootDirAbs dbSchemaChecksumInProjectRootDir = do
-  dbSchemaExists <- doesFileExist dbSchemaFp
-  when dbSchemaExists $ do
-    checksum <- hexToString <$> checksumFromFilePath dbSchemaFp
-    writeFile dbSchemaChecksumFp checksum
+  whenM (doesFileExist dbSchemaFile) $ do
+    checksum <- hexToString <$> checksumFromFilePath dbSchemaFile
+    IOUtil.writeFile dbSchemaChecksumFile checksum
   where
-    dbSchemaFp = SP.fromAbsFile $ genProjectRootDirAbs SP.</> dbSchemaFileInProjectRootDir
-    dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDirAbs SP.</> dbSchemaChecksumInProjectRootDir
+    dbSchemaFile = genProjectRootDirAbs </> dbSchemaFileInProjectRootDir
+    dbSchemaChecksumFile = genProjectRootDirAbs </> dbSchemaChecksumInProjectRootDir
 
-removeDbSchemaChecksumFile :: Path' Abs (Dir ProjectRootDir) -> Path' (Rel ProjectRootDir) File' -> IO ()
-removeDbSchemaChecksumFile genProjectRootDirAbs dbSchemaChecksumInProjectRootDir =
-  let dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDirAbs SP.</> dbSchemaChecksumInProjectRootDir
-   in removeFile dbSchemaChecksumFp
+removeDbSchemaChecksumFile ::
+  DbSchemaChecksumFile f =>
+  Path' Abs (Dir ProjectRootDir) ->
+  Path' (Rel ProjectRootDir) (File f) ->
+  IO ()
+removeDbSchemaChecksumFile genProjectRootDirAbs dbSchemaChecksumInProjectRootDir = removeFile dbSchemaChecksumFp
+  where
+    dbSchemaChecksumFp = genProjectRootDirAbs </> dbSchemaChecksumInProjectRootDir
+
+generatePrismaClients :: Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
+generatePrismaClients projectRootDir = do
+  generateResult <- liftA2 (>>) generatePrismaClientForServer generatePrismaClientForWebApp projectRootDir
```

Comment:
It's right there on the edge :smile:
See my comment here: https://github.com/wasp-lang/wasp/pull/962#discussion_r1098700396

It's the same thing as `liftM2`, but `liftM2` will get deprecated (since all Monads are Applicatives).